diofant/integrals/risch.py

```diff
@@ -965,29 +965,29 @@ def hermite_reduce(a, d, DE):
 
     dd = derivation(d, DE)
     dm = gcd(d, dd).as_poly(DE.t)
-    ds, r = d.div(dm)
+    ds, _ = d.div(dm)
 
     while dm.degree(DE.t) > 0:
-
         ddm = derivation(dm, DE)
         dm2 = gcd(dm, ddm)
-        dms, r = dm.div(dm2)
+        dms, _ = dm.div(dm2)
         ds_ddm = ds*ddm
         ds_ddm_dm, r = ds_ddm.div(dm)
 
         b, c = gcdex_diophantine(-ds_ddm_dm.as_poly(DE.t), dms.as_poly(DE.t), a.as_poly(DE.t))
         b, c = b.as_poly(DE.t), c.as_poly(DE.t)
 
         db = derivation(b, DE).as_poly(DE.t)
-        ds_dms, r = ds.div(dms)
+        ds_dms, _ = ds.div(dms)
         a = c.as_poly(DE.t) - (db*ds_dms).as_poly(DE.t)
 
         ga = ga*dm + b*gd
         gd = gd*dm
         ga, gd = ga.cancel(gd, include=True)
         dm = dm2
 
-    d = ds
+        d = ds
```

Comment:
Maybe this was just a wrong indentation.

Reply:
This may take care of the case where the denominator is square-free, but I am not sure of the general case. I suspect that the issue could reappear in the computation of dm2. I would prefer a solution where the polynomials have coefficients in a field.